pandas/core/reshape/encoding.py

```diff
@@ -175,7 +175,9 @@ def check_len(item, name):
 
         # validate separators
         if isinstance(prefix_sep, str):
-            prefix_sep = itertools.cycle([prefix_sep])
+            # error: Incompatible types in assignment (expression has type
+            # "cycle[str]", variable has type "str")
+            prefix_sep = itertools.cycle([prefix_sep])  # type: ignore[assignment]
```

Comment:
Should the variable just be re-assigned to avoid the ignore?

Reply:
I found a workaround: this function works with any `Iterable[str]` not just `list[str]`. `cycle[str]` is also `Iterable[str]` :)